src/Aspire.Hosting/Dashboard/DashboardTokenProvider.cs

```diff
@@ -0,0 +1,37 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text;
+using Aspire.Hosting.Utils;
+using Microsoft.Extensions.Options;
+
+namespace Aspire.Hosting.Dashboard;
+
+internal interface IDashboardTokenProvider
+{
+    public string BrowserToken { get; }
+    public string OltpToken { get; }
+    public string ResourceServerToken { get; }
+}
+
+internal class DashboardTokenProvider : IDashboardTokenProvider
+{
+    public DashboardTokenProvider(IOptions<DashboardAuthenticationOptions> dashboardAuthenticationOptions)
+    {
+        BrowserToken = dashboardAuthenticationOptions.Value.BrowserToken ?? GenerateToken();
+        OltpToken = GenerateToken();
+        ResourceServerToken = GenerateToken();
+    }
+
+    private static string GenerateToken()
+    {
+        var rawToken = PasswordGenerator.Generate(24, true, true, true, true, 0, 0, 0, 0);
```

Comment:
Named arguments for the literals here would make this easier to read without an IDE.

Reply:
I'll go one better and default the values so no one needs to think about it :)